src/Services/LoggingService.php

```diff
@@ -116,17 +126,35 @@ private function getPlentyMarketLoggerInstance(string $method, string $reference
   }
 
   /**
-   * Maps data from the he associative array that Wayfair provides to the PlentyMarkets logging API's inputs
+   * Maps data from the associative array that Wayfair provides to the PlentyMarkets logging API's inputs
    *
    * @param $loggingInfo
    *
    * @return array
    */
   public function extractVars($loggingInfo): array {
+    $externalLogs = pluginApp(ExternalLogs::class);
+    $clientID = $this->configHelper->getClientId();
+    $shortMessage = [];
+
     $additionalInfo = $loggingInfo['additionalInfo'] ?? [];
     $method = $loggingInfo['method'] ?? null;
     $referenceType = $loggingInfo['referenceType'] ?? null;
     $referenceValue = (int) $loggingInfo['referenceValue'] ?? null;
+
+    if (strlen(json_encode($loggingInfo)) > self::STRING_LIMIT) {
+      $logForKibana = [];
+      $shortMessage['message'] = self::LOG_KEY_MESSAGE_TO_LONG . $clientID . '-' . date('M d Y H:i:s');
+      $additionalInfo = $shortMessage;
+      $logForKibana['message'] = self::LOG_KEY_MESSAGE_TO_LONG . $clientID . '-' . date('D, d M Y H:i:s');
+      $logForKibana['details'] = $loggingInfo;
+      $externalLogs->addErrorLog(json_encode($logForKibana));
+      if (count($externalLogs->getLogs())) {
+        /** @var LogSenderService $logSenderService */
+        $logSenderService = pluginApp(LogSenderService::class);
+        $logSenderService->execute($externalLogs->getLogs());
```

Comment:
always do this in a `finally` block so that logs are not lost due to exceptions.